plone/app/contenttypes/browser/link_redirect_view.py

```diff
@@ -111,10 +112,16 @@ def absolute_target_url(self):
             if "resolveuid" in url:
                 uid = url.split("/")[-1]
                 obj = uuidToObject(uid)
+
+                portal_state = getMultiAdapter(
+                    (self.context, self.request), name="plone_portal_state"
+                )
+                portal_url = portal_state.portal_url()
+
                 if obj:
                     url = "/".join(obj.getPhysicalPath()[2:])
```

Comment:
I might be wrong, but if this `[:2]` is to remove something like "/app/Plone", then it is not valid...
You can have Plone site inside other objects, e.g. a `Folder`.
e.g. the path might be `/app/folder/Plone` or `/app/folder/subfolder/Plone`.

Reply:
Consider also VHM rules, and the use of `_vh_` (for example: 
```apache
    location /zope {
        proxy_pass http://localhost:8080/VirtualHostBase/https/site.com/VirtualHostRoot/_vh_zope;
    }
```